confd/etc/calico/confd/templates/bird.cfg.template

```diff
@@ -107,42 +46,30 @@ protocol kernel {
 
 # Watch interface up/down events.
 protocol device {
-{{- template "LOGGING"}}
+{{- template "LOGGING" $config}}
   scan time 2;    # Scan interfaces every 2 seconds
 }
 
-{{- $ignored_interfaces := ""}}
-{{- $node_ignored_interfaces_key := printf "/bgp/v1/host/%s/ignored_interfaces" (getenv "NODENAME")}}
-{{- if exists $node_ignored_interfaces_key }}{{$ignored_interfaces = getv $node_ignored_interfaces_key}}
-{{- else if exists "/bgp/v1/global/ignored_interfaces" }}{{$ignored_interfaces = getv "/bgp/v1/global/ignored_interfaces"}}
-{{- end}}
-
 protocol direct {
-{{- template "LOGGING"}}
-{{- if ne "" $ignored_interfaces}}{{$ifaces := split $ignored_interfaces ","}}
-  interface {{range $ifaces}}-"{{.}}", {{end}}-"cali*", -"kube-ipvs*", "*";
-{{- else}}
-  interface -"cali*", -"kube-ipvs*", "*"; # Exclude cali* and kube-ipvs* but
+{{- template "LOGGING" $config}}
+  interface {{$config.DirectInterfaces}};{{if eq $config.DirectInterfaces `-"cali*", -"kube-ipvs*", "*"`}} # Exclude cali* and kube-ipvs* but
                                           # include everything else.  In
                                           # IPVS-mode, kube-proxy creates a
                                           # kube-ipvs0 interface. We exclude
                                           # kube-ipvs0 because this interface
                                           # gets an address for every in use
                                           # cluster IP. We use static routes
                                           # for when we legitimately want to
-                                          # export cluster IPs.
-{{- end}}
+                                          # export cluster IPs.{{end}}
 }
 
-{{if eq "" ($node_ip)}}# IPv4 disabled on this node.
-{{else}}{{$node_as_key := printf "/bgp/v1/host/%s/as_num" (getenv "NODENAME")}}
+{{if eq "" $config.NodeIP}}# IPv4 disabled on this node.
+{{else}}
 # Template for all BGP clients
 template bgp bgp_template {
-{{- $as_key := or (and (exists $node_as_key) $node_as_key) "/bgp/v1/global/as_num"}}
-{{- $node_as_num := getv $as_key}}
-{{- template "LOGGING"}}
+{{- template "LOGGING" $config}}
   description "Connection to BGP peer";
-  local as {{$node_as_num}};
+  local as {{$config.AsNumber}};
```

Comment:
nit, I think this should be `ASNumber`, because that looks better to me, and for consistency with `NodeIP`